### PR TITLE
Added ExportTimestampEnrichmentExporter

### PR DESCRIPTION
**Pull request by kkruk-sumo**
To learn more about why this PR is needed please read https://github.com/open-telemetry/opentelemetry-specification/issues/1270 .

Usage:
```javascript
const exporter = new ExportTimestampEnrichmentExporter(originalExporter);
```

New exporter has been introduced to make sure that this solution will work with different exporters like OTLP.

### CODE REVIEW

**Review by pmm-sumo — APPROVED** [review]
Looks good. Perhaps unit test could be added?

**Review by piecioshka — APPROVED** [review]
Nice! 👏

**Review by mateusz-sumo — APPROVED** [review]

**Review by mateusz-sumo — APPROVED** [review]

File: src/opentelemetry-export-timestamp-enrichment/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { CanonicalCode, SpanKind, TraceFlags } from '@opentelemetry/api';
+import { Resource } from '@opentelemetry/resources';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { ExportTimestampEnrichmentExporter } from '.';
+
+beforeAll(() => {
+  jest.useFakeTimers('modern').setSystemTime(new Date('2020-10-01').getTime());
```
Comment (piecioshka) [inline src/opentelemetry-export-timestamp-enrichment/index.spec.ts:7]: 🚀

**Review by piecioshka — APPROVED** [review]
Rocks! 🚀

**Pull request merged**